pkg/flowcli/project/transaction.go

```diff
@@ -0,0 +1,243 @@
+package project
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/onflow/flow-go-sdk/templates"
+
+	"github.com/onflow/flow-cli/pkg/flowcli"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/flow-go-sdk"
+
+	"github.com/onflow/flow-cli/pkg/flowcli/util"
+)
+
+type signerRole string
+
+const (
+	SignerRoleAuthorizer signerRole = "authorizer"
+	SignerRoleProposer   signerRole = "proposer"
+	SignerRolePayer      signerRole = "payer"
+)
+
+func NewTransaction() *Transaction {
+	return &Transaction{
+		tx:        flow.NewTransaction(),
+		contracts: []templates.Contract{},
+	}
+}
+
+type Transaction struct {
+	signer    *Account
+	role      signerRole
+	proposer  *Account
+	payer     flow.Address
+	tx        *flow.Transaction
+	contracts []templates.Contract
```

Comment:
Contracts feels out of place here, given that it's only used by transactions that deploy contracts. What's the overall use case for the `Transaction` struct? It feels a bit like a bloated class to me. lt takes care of various actions related to preparing and signing a transaction, but it also has implementation details of specific types of account and contract transactions.

Reply:
I agree about the contracts. 
Well, the purpose or idea of the Transaction object was that I wouldn't have a Send service function that is long and has an implementation of creating transactions that could be extracted somewhere else. Do you have any other idea how to achieve this?